tests/test_ser.py

```diff
@@ -95,6 +95,47 @@ def fail(x):
         s.serialize_value(Q())
 
 
+def test_subclass():
+    class SubclassedUUID(uuid.UUID):  # such as asyncpg.pgproto.pgproto.UUID
+        pass
+
+    _uuid = s.serialize_value(SubclassedUUID("939d4cc9-830d-4cca-bd74-3ec3d541a9b3"))
+    assert _uuid == "uuid:939d4cc9-830d-4cca-bd74-3ec3d541a9b3"
+
+
+def test_subclass_mro_order():
+    class A:
+        pass
+
+    class B:
+        pass
+
+    class C(A, B):
+        pass
+
+    class D(C, B):
+        pass
+
+    class CaughtExceptionA(Exception):
+        pass
+
+    class CaughtExceptionB(Exception):
+        pass
+
+    def a(x):
+        return "a"
+
+    def b(x):
+        return "b"
+
+    s.register_type(A, code="suba", deserializer=a, serializer=a)
+    s.register_type(B, code="subb", deserializer=b, serializer=b)
```

Comment:
using the same function as serializer/deserializer is a bit confusing here - how about we just leave the deserializer out entirely and only test serialization here, since mro isn't a concern for deserialization.